core/src/main/java/edu/wpi/grip/core/Pipeline.java

```diff
@@ -69,6 +68,15 @@ public void clear() {
         return Collections.unmodifiableSet(this.connections);
     }
 
+    /*
+     * @return The current per-project settings.  This object may become out of date if the settings are edited
+     * by the user, so objects requiring a preference value should also subscribe to {@link ProjectSettingsChangedEvent}
+     * to get updates.
+     */
+    public ProjectSettings getProjectSettings() {
+        return settings;
```

Comment:
I understand, and I think that the value added by that is very little compared to the maintenance effort required

Since we control all of the code that accesses `ProjectSettings`, and it's currently only a few places, it's pretty easy to verify that we're not accidently updating it when we should.  If fact, only two of those places (the settings dialog and the deserialization stuff) update it, and I don't see why we would change the settings anywhere else.

(this is a "will not fix", btw)